### PR TITLE
Field report: CardCoins rbf and batching

### PR DESCRIPTION
**Pull request by bitschmidty**
- [x] Official CardCoins blog link added to article

### CODE REVIEW

File: _includes/articles/cardcoins-rbf-batching.md [hunk 1]
```diff
@@ -33,11 +33,16 @@ each customer requests a withdrawal, an output is added to the transaction in
 the mempool. This transaction continues to be updated until it confirms or
 reaches some other local optimum.
 
-There are many strategies to this type of additive RBF batching. At CardCoins we
-took a safety-first approach to our implementation (with the help of Matthew
-Zipkin), the nuances of which we detailed in a more detailed blog post, RBF
-Batching at CardCoins: Diving into the Mempool's Dark Reorg Forest.
+There are many strategies to this type of additive RBF batching. At [CardCoins][] we
+took a safety-first approach to our implementation (with the help of [Matthew
+Zipkin][]), the nuances of which we detailed in a more detailed blog post, [RBF
```
Comment (CardCoins) [inline _includes/articles/cardcoins-rbf-batching.md:38]: Minor edit, remove the extra 'detailed': "the nuances of which we detailed in the blog post,"

File: _includes/articles/cardcoins-rbf-batching.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+When combining these two tools, a service provider unlocks new functionality but
+is similarly exposed to novel forms of complexity. In the base case, combining
+RBF and a single, static batch carries a simple combination of the complexities
+that RBF and batching carry discretely. However, when you combine RBF and
+"additive batching," emergent edge cases and dangerous failure scenarios present
+themselves.
+
+Additive RBF batching is a transaction construction protocol that leverages the
+rules of BIP-125 to allow a transactor to introduce new outputs (and confirmed
+inputs) to a transaction in the mempool. This enables an enterprise to give
```
Comment (jnewbery) [inline _includes/articles/cardcoins-rbf-batching.md:29]: Maybe this definition or something similar could be put in the opening paragraph to introduce the subject.

**Review by jnewbery — COMMENTED** [review]
Looks good. I pushed a commit with various changes. Let me know what you think.

File: _includes/articles/cardcoins-rbf-batching.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Each tool, while powerful, has its own complexities and nuances. For example,
+batching customer withdrawals may save on fees for the enterprise, but will likely make
+[child pays for parent][topic cpfp] (CPFP) uneconomical for a customer who wishes to speed up the transaction.
+Similarly, RBF is useful for an enterprise who takes a fee-underbidding strategy
+(their initial transaction broadcast starts at a low fee, and is slowly bid
+upwards), but it exposes their customers to [potential confusion][rbf blog] as their
+withdrawal transaction updates in their wallet. It would also be messy for the
+customer to spend from this transaction while it remains unconfirmed, as the
+enterprise will have to account for this child spend when attempting to replace
```
Comment (harding) [inline _includes/articles/cardcoins-rbf-batching.md:16]: Nit: I think this is more specific about the immediate burden placed on the enterprise.

```suggestion
enterprise will have to pay for this child spend when attempting to replace
```

File: _includes/articles/cardcoins-rbf-batching.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+the parent. Even worse, the enterprise may have a withdrawal [pinned][pinning] by another
+service that the customer withdrew their funds directly to.
```
Comment (harding) [inline _includes/articles/cardcoins-rbf-batching.md:18]: Nitpick.

```suggestion
service which received the customer's withdrawal.
```

File: _includes/articles/cardcoins-rbf-batching.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{:.post-meta}
+*by [CardCoins][]*
+
+[Replace By Fee][topic rbf] (RBF, BIP-125) and [batching][payment batching] are two important tools for any enterprises
```
Comment (harding) [inline _includes/articles/cardcoins-rbf-batching.md:4]: Nit: use our normal BIP styling; the consistency is especially helpful to me when I'm `git grep`ing for things we previously wrote.

```suggestion
[Replace By Fee][topic rbf] (RBF, BIP125) and [batching][payment batching] are two important tools for any enterprises
```

Comment (harding) [reply _includes/articles/cardcoins-rbf-batching.md:4]: Also needs edit later in the article (not important for git grep, but important for intra-article consistency).

**Review by harding — COMMENTED** [review]
A few nitpicks.

**Comment by bitschmidty** [comment]
Resolved all comments except: https://github.com/bitcoinops/bitcoinops.github.io/pull/582#discussion_r643064073

@CardCoins do you have an opinion on how youd like to address this if at all?

**Comment by CardCoins** [comment]
@bitschmidty Sorry just getting back from the conference, will have a close look tomorrow. Thanks all for the reviews.

**Comment by bitschmidty** [comment]
Thanks @CardCoins ! 

We plan to publish this post on June 16, along with our weekly newsletter.

One more thing needed to wrap up this summary post would be the url of the 'Mempool's Dark Reorg Forest', when published.

**Comment by CardCoins** [comment]
@bitschmidty The final url will be: blog.cardcoins.co/rbf-batching-at-cardcoins-diving-into-the-mempool-s-dark-reorg-forest.

File: _includes/articles/cardcoins-rbf-batching.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{:.post-meta}
+*by [CardCoins][]*
+
```
Comment (CardCoins) [inline _includes/articles/cardcoins-rbf-batching.md:3]: If we want to include an short definition at the opening of the blog post, here is a suggestion: 

"Additive batching" is a scheme whereby additional outputs are included to unconfirmed transactions in the mempool. This field report outlines efforts CardCoins has in implementing a reorg and DoS safe implementation of such a scheme.

Will let you guys ultimately decide on its inclusion. I could go either way.

**Comment by jnewbery** [comment]
The new intro looks good, but it makes some of the text in the later paragraph redundant. I've pushed a commit that makes a few changes to reduce that redundancy.

**Pull request closed without merging**